services/ui-src/src/components/fields/ChoiceListField.tsx

```diff
@@ -102,14 +110,15 @@ export const ChoiceListField = ({
               if (child.props?.choices) {
                 child.props.choices.forEach((choice: FieldChoice) => {
                   choice.checked = false;
-                  form.setValue(child.id, []);
                 });
+                form.setValue(child.id, []);
                 clearUncheckedNestedFields(child.props.choices);
               }
               break;
             default:
               child.props = { ...child.props, clear: true };
               form.setValue(child.id, "");
+              form.unregister(child.id);
```

Comment:
Errors need to clear if a text box is removed from the page and the only way to properly do it is by unregistering. And why not for choicelists? Because if I do this for choicelists it breaks consistency with nested choicelist. This is the one time I'm gonna ask you to trust me that while awkward I think this provides a more consistent user experience.